tests/test_algorithms.py

```diff
@@ -0,0 +1,30 @@
+from collections import OrderedDict
+
+import theano
+from numpy.testing import assert_allclose
+from theano import tensor
+
+from blocks.extras.algorithms import BasicNesterovMomentum, NesterovMomentum
+from blocks.utils import shared_floatx
+
+
+def test_basic_nesterov_momentum():
+    a = shared_floatx([3, 4])
+    cost = (a ** 2).sum()
+    steps, updates = BasicNesterovMomentum(0.5).compute_steps(
+        OrderedDict([(a, tensor.grad(cost, a))]))
+    f = theano.function([], [steps[a]], updates=updates)
+    assert_allclose(f()[0], [9., 12.])
```

Comment:
Can you compile another function, which computes the update like in equations (1)-(2) of the paper and compare here that simplified NAG returns the same result?